### PR TITLE
Explosion Block (A hidden stat on things like reinforced walls) is more effective against heavy and light tiers of explosions

### PR DESCRIPTION
**Pull request by moocowswag**
## About The Pull Request

As of right now if a 5/10/20 bomb goes through a reinforced wall it has its ranges subtracted by a flat value of 2 (explosion block is a flat subtraction to each tier) and essentially gets turned into a 3/8/18.

This pr just buffs the effect of explosion block against heavy and light. The modifier is 1.5x against heavy and 2.5x against light (I might nerf 2.5x down to 2x) I dont want to nullify shockwaves because they should always be impactful at weakening the station.

In the same situation a 5/10/20 bomb goes through a reinforced wall, it has its ranged subtracted and ends up 3/7/15

Here is an image of a max cap (5/10/20, do note that currently tg server config is like 8/16/32 or something crazy) in tram sci, notice there is roughly 2 layers of reinforced walls on the right and it makes it through the first layer and hits the second, it would likely destroy the entire first layer of reinforced walls and remove several tiles from the second making a second explosion able to go into genetics entirely unmitigated.
image

In practice the 2nd layer of walls remains miraculously unscathed due to light explosions having an rng diceroll on destroying reinforced walls

image

A second blast still gets through since the 1st layer is no longer there to protect it.

image

Do note on live servers it would be much bigger due to max cap being 8/16/32 , going through 1 reinforced walls would result in 6/13/27 which is still far larger than the 5/10/20 shown.

## Why It's Good For The Game

Its probably not the best approach to fixing the problem and I probably made it too good at mitigating light but..

As of right now reinforced walls are so weak at stopping or mitigating explosions that they are spammed alittle bit more than they should be on a lot of maps and the nukie ship has a laughably silly explosion block of 20!!! Just to be able to make it somewhat safe.

Every once in awhile atmos becomes a hot topic for debate because of some new form of spammable explosive that is getting nerfed because of the way they can easily wipe the station but we never took a look at how explosions are done, if this pr is merged it won't kill those types of gimmicks but it will necessitate actually thinking about how you plan to bomb the station since now you will need to repeatedly bomb to get through hardpoints (remember even if a reinforced wall mitigates an explosion it can be dismantled by it and let subsequent explosions through) or simply position those bombs directly inside of areas of interest

Another important note is that rooms that map makers intend to be explosive resistant should be able to better contain 1 explosion so less situations of everyone in sci getting cucked by toxins messing up (unless its so bad that they bomb themselves twice)

If this pr is merged I would suggest keeping an eye on heavy firelocks, they are more expensive than reinforced walls but allow movement through them which means they are a lot more practical for spamming explosive block.

I personally wanted to make foam and other materials good at uniquely blocking different tiers of explosions but I recall being told that was a little bit too complicated, if this pr doesn't go well I might try that.
## Changelog
:cl:

balance: Reinforced sections of station hull have gotten better at mitigating explosive shockwaves.
/:cl:

### CODE REVIEW

**Comment by cnleth** [comment]
> now you will need to repeatedly bomb to get through hardpoints (remember even if a reinforced wall mitigates an explosion it can be dismantled by it and let subsequent explosions through)

Why do you think punishing people who can't afford bombing more than once (e.g. a traitor who bought a syndiebomb and doesn't care about toxins) is good for the game?
The requirement of using more explosives or placing them differently to break walls seems like the opposite of discouraging spamming explosives, which seems to be the "problem" you want to fix with this PR?

**Comment by moocowswag** [comment]
> > now you will need to repeatedly bomb to get through hardpoints (remember even if a reinforced wall mitigates an explosion it can be dismantled by it and let subsequent explosions through)
> 
> Why do you think punishing people who can't afford bombing more than once (e.g. a traitor who bought a syndiebomb and doesn't care about toxins) is good for the game? The requirement of using more explosives or placing them differently to break walls seems like the opposite of discouraging spamming explosives, which seems to be the "problem" you want to fix with this PR?

The problem is that explosion block is useless and this pr wont discourage traitors from bombing the places they tend to bomb anyway, a lot of good places to bomb are not reinforced by map makers and the end result should be that....

1. Workplace accidents that map makers tried to mitigate are better mitigated (again the most armored place is probably toxins)
2. Adds pre-emptive counterplay to explosives...remember explosives are 360 degrees plasteel is expensive, and very few places are inherently reinforced and capable of fully containing or protecting from a blast
3. Makes bomb placement less "okay anything in this radius is gonna be wiped" and more like "okay anything in this radius except stuff from that angle is gonna be wiped. 

I would actually argue its a buff to the person who is in control of the bombs and raises the skill ceiling to both using them and playing against them, rather than it being purely about the radius of the explosion, cover comes into play and now the door is opened for situations where you can still stay in the blast radius, possibly baiting people to also stay in it, but you are in cover and they arnt and they dont know that.

**Comment by moocowswag** [comment]
> Why do you think punishing people who can't afford bombing more than once (e.g. a traitor who bought a syndiebomb and doesn't care about toxins) is good for the game? 

Also i think that piece of crap could use a price reduction or blast radius increase considering how easy it is to just hack into toxins and grab your own max cap that you can instantly and discreetly detonate without spending special antag resources.

**Comment by breadcrombs** [comment]
this is actually a secert pr to buff reinforced firelocks for your flavor of the month esoteric bci memery

**Comment by moocowswag** [comment]
> this is actually a secret pr to buff reinforced firelocks for moocow's flavor of the month esoteric bci memery

I wanna nerf them to 1 along with the nukie ship walls to like 7 and maybe plastitanium stuff to 3.

**Comment by Higgin** [comment]
good change

File: code/controllers/subsystem/explosions.dm [hunk 1]
```diff
@@ -121,25 +121,26 @@ SUBSYSTEM_DEF(explosions)
...
-		else if (dist < heavy)
+		else if (dist + block * 1.5 < heavy)
 			explode.color = "yellow"
-			explode.maptext = MAPTEXT("[dist]")
-		else if (dist < light)
+			explode.maptext = MAPTEXT("[dist + block * 1.5]")
+		else if (dist + block * 2.5 < light)
 			explode.color = "blue"
-			explode.maptext = MAPTEXT("[dist]")
+			explode.maptext = MAPTEXT("[dist + block * 2.5]")
```
Comment (Ghommie) [inline code/controllers/subsystem/explosions.dm:143]: Could you use defines (def/undef within the file) for these multipliers?

**Review by Ghommie — APPROVED** [review]
Let's give this a twirl. If the multipliers for the explosion block are too high, we can lower them.

File: code/controllers/subsystem/explosions.dm (hunk 1 above)
Comment (moocowswag) [reply code/controllers/subsystem/explosions.dm:143]: as in you want a variable to easily change them all from?

Comment (moocowswag) [reply code/controllers/subsystem/explosions.dm]: i think i did it

**Pull request merged**